components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/model/ServiceURL.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.core.model;
+
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+
+import java.util.Map;
+
+/**
+ * Service URL representation.
+ */
+public class ServiceURL {
+
+    private String protocol;
+    private String hostName;
+    private Integer port;
+    private String proxyContext;
+    private String webContext;
+    private String urlContext;
+    private Map<String, String> parameters;
+    private Map<String, String> fragments;
+
+    public ServiceURL(String protocol, String hostName, Integer port, String proxyContext, String webContext,
+                      String urlContext, Map<String, String> parameters, Map<String, String> fragments) {
+
+        this.protocol = protocol;
+        this.hostName = hostName;
+        this.port = port;
+        this.proxyContext = proxyContext;
+        this.webContext = webContext;
+        this.urlContext = urlContext;
+        this.parameters = parameters;
+        this.fragments = fragments;
+    }
+
+    public String getProtocol() {
+
+        return protocol;
+    }
+
+    public String getHostName() {
+
+        return hostName;
+    }
+
+    public Integer getPort() {
+
+        return port;
+    }
+
+    public String getProxyContext() {
+
+        return proxyContext;
+    }
+
+    public String getWebContext() {
+
+        return webContext;
+    }
+
+    public String getUrlContext() {
+
+        return urlContext;
+    }
+
+    public Map<String, String> getParameters() {
```

Comment:
We should not return the mutable map.
Make map unmodifiable. Also add a method get the value from the parameter key and a method to get just parameter names